IPython/core/application.py

```diff
@@ -215,7 +215,7 @@ def excepthook(self, etype, evalue, tb):
             return crashhandler.crash_handler_lite(etype, evalue, tb)
     
     def _ipython_dir_changed(self, name, old, new):
-        if old is not None:
+        if old is not None and old is not Undefined:
```

Comment:
ok, makes sense. There may be other cases affected by this. We'll have to keep an eye out.